packages/react-native/ReactAndroid/src/main/java/com/facebook/react/LazyReactPackage.kt

```diff
@@ -0,0 +1,140 @@
+                    override fun hasNext(): Boolean {
+                        return position < nativeModules.size
+                    }
+
+                    override fun next(): ModuleHolder {
+                        val moduleSpec = nativeModules[position++]
+                        val name = moduleSpec.name
+
+                        return reactModuleInfoMap[name].let moduleHolder@{ reactModuleInfo ->
+                            if (reactModuleInfo == null) {
+                                val module: NativeModule
+                                ReactMarker.logMarker(
+                                    ReactMarkerConstants.CREATE_MODULE_START,
+                                    name,
+                                )
+                                try {
+                                    module = moduleSpec.provider.get()
+                                } finally {
+                                    ReactMarker.logMarker(ReactMarkerConstants.CREATE_MODULE_END)
+                                }
+                                return@moduleHolder ModuleHolder(module)
+                            } else {
+                                return@moduleHolder ModuleHolder(
+                                    reactModuleInfo,
+                                    moduleSpec.provider,
+                                )
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param reactContext react application context that can be used to create modules
+     * @return list of module specs that can create the native modules
+     */
+    public abstract fun getNativeModules(reactContext: ReactApplicationContext?): List<ModuleSpec>
+
+    /**
+     * @param reactContext react application context that can be used to create modules
+     * @return [<] to register
+     */
+    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
+        val modules: MutableList<NativeModule> = ArrayList()
+        for (holder in getNativeModules(reactContext)) {
+            var nativeModule: NativeModule
+            SystraceMessage.beginSection(TRACE_TAG_REACT, "createNativeModule").flush()
+            ReactMarker.logMarker(ReactMarkerConstants.CREATE_MODULE_START, holder.name)
+            try {
+                nativeModule = holder.provider.get()
+            } finally {
+                ReactMarker.logMarker(ReactMarkerConstants.CREATE_MODULE_END)
+                SystraceMessage.endSection(TRACE_TAG_REACT).flush()
+            }
+            modules.add(nativeModule)
+        }
+        return modules
+    }
+
+    /**
+     * @param reactContext react application context that can be used to create View Managers.
+     * @return list of module specs that can create the View Managers.
+     */
+    public fun getViewManagers(reactContext: ReactApplicationContext?): List<ModuleSpec> {
```

Comment:
Let's try to keep the same nullability for the arguments as in the original implementation

```suggestion
    public fun getViewManagers(reactContext: ReactApplicationContext): List<ModuleSpec> {
```